### PR TITLE
Add DIDs to config file

### LINKED ISSUE #70: Create valid Gaia-X Credenials according to Gaia-X Tagus release
GX Credential Generator generates one huge json-ld file with ALL information about SCS cluster, currently. In favor to flexibility and maintenance, each credential should be serialized in its own file and referenced by its [DID](https://www.w3.org/TR/did-core/) in other credentials. 

This issue contains two tasks:

- [x]  Implement DID creator in python similar to Gaia-X Lab [DID Creator](https://gitlab.com/gaia-x/lab/did-web-generator) written in JavaScript. See https://github.com/SovereignCloudStack/scs-did-creator/
- [x]  Add DID of SCS cluster and provider to generated GX Credentials, - [scs-did-creator#3](https://github.com/SovereignCloudStack/scs-did-creator/pull/3) 
- [x]  Create Gaia-X compliant credential for CSP's as `LegalPerson` and compliant credential for OpenStack cloud as `VirtualMachineServiceOffering`-  #88 
- [ ]  ~Figure out how GX Credential can be retrieved from a given DID.~ --> moved to a separate issue (#102 ), for now, all credentials are in one file.

### PR DESCRIPTION
**Pull request by anjastrunk**
related to #70

### CODE REVIEW

File: generator/discovery/openstack/openstack_discovery.py [hunk 1]
```diff
@@ -4,32 +4,95 @@
...
+            ),
+            formatType=self.config.get_value(
+                [
+                    const.CONFIG_IAAS,
+                    const.CONFIG_IAAS_DATA_EXPORT,
+                    const.CONFIG_IAAS_DATA_EXPORT_FORMAT_TYPE,
+                ]
+            ),
+        )
+        serivce_tac = []
```
Comment (martinmo) [inline generator/discovery/openstack/openstack_discovery.py:65]:
```suggestion
        service_tac = []
```

File: generator/discovery/openstack/openstack_discovery.py [hunk 2]
```diff
@@ -4,32 +4,95 @@
...
+        for url in self.config.get_value(
+                [const.CONFIG_IAAS, const.CONFIG_IAAS_T_AND_C]
+        ):
+            httpResponse = requests.get(url)
+            if httpResponse.status_code == 200:
+                content = httpResponse.text
+                serivce_tac.append(
```
Comment (martinmo) [inline generator/discovery/openstack/openstack_discovery.py:72]:
```suggestion
                service_tac.append(
```

File: generator/discovery/openstack/openstack_discovery.py [hunk 3]
```diff
@@ -4,32 +4,95 @@
...
+                        url=url, hash=sha256(content.encode("utf-8")).hexdigest()
+                    )
+                )
+            else:
+                raise HTTPError(
+                    "Cloud not retrieve terms and conditions from '"
+                    + url + "'. HTTP Status code: " + str(httpResponse.status_code)
+                )
+
+        if len(serivce_tac) == 0:
```
Comment (martinmo) [inline generator/discovery/openstack/openstack_discovery.py:83]:
```suggestion
        if len(service_tac) == 0:
```

File: generator/discovery/openstack/openstack_discovery.py [hunk 4]
```diff
@@ -4,32 +4,95 @@
...
+                + const.CONFIG_IAAS + "." + const.CONFIG_IAAS_T_AND_C
+            )
+
+        return VirtualMachineServiceOffering(
+            providedBy=self.config.get_value([const.CONFIG_CSP, const.CONFIG_DID]),
+            dataAccountExport=data_export_account,
+            servicePolicy=self.config.get_value(
+                [const.CONFIG_IAAS, const.CONFIG_IAAS_SERVICE_POLICY]
+            ),
+            serviceOfferingTermsAndConditions=serivce_tac,
```
Comment (martinmo) [inline generator/discovery/openstack/openstack_discovery.py:95]:
```suggestion
            serviceOfferingTermsAndConditions=service_tac,
```

File: tests/test_cli.py [hunk 5]
```diff
@@ -3,81 +3,45 @@
...
-        "uuid_validity": "last-3",
-        "patchlevel": "1.5.2",
-        "license_required": True,
-        "license_included": False,
-        "subscription_required": True,
-        "subscription_included": False,
-        "maintained_until": "2024-05-31",
+EXPECTED_RESULT = {
+    "@context": {
+        "gx": "https://w3id.org/gaia-x/ONTOLOGY_VERSION#",
```
Comment (martinmo) [inline tests/test_cli.py:13]: Question: `https://w3id.org/gaia-x/ONTOLOGY_VERSION#` also appears in the generated `gx_schema.py` -> is `ONTOLOGY_VERSION` a placeholder for some actual version like `v1`, `v2`?

**Review by martinmo — APPROVED** [review]
Just found typos, apart from that looks good.

Out of curiosity: how is the `gx_schema.py` file generated? Does it make sense to add instructions/tooling for that to this repository? For example, which tools and inputs are needed and where to get them from?

File: tests/test_cli.py (hunk 5 above)
Comment (anjastrunk) [reply tests/test_cli.py:13]: Yes, it is and there is still a typo in. It should be `https://w3id.org/gaia-x/deployment#`. I will change this.

**Comment by anjastrunk** [comment]
> Just found typos, apart from that looks good.
> 
> Out of curiosity: how is the `gx_schema.py` file generated? Does it make sense to add instructions/tooling for that to this repository? For example, which tools and inputs are needed and where to get them from?

This file is auto-generated bx linkml. I will extend doc to point this out.

**Pull request merged**